sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -64,6 +66,7 @@ public OfferEvaluator(StateStore stateStore, OfferRequirementProvider offerRequi
             }
 
             if (!failureNotifications.isEmpty()) {
+                recommendations.clear();
```

Comment:
Could also just make a new `recommendations` at the head of each loop run, up to you.